src/main/java/duke/task/Deadline.java

```diff
@@ -0,0 +1,81 @@
+package duke.task;
+import duke.task.Task;
+
+import java.time.LocalDateTime;
+import java.time.format.DateTimeFormatter;
+
+
+/**
+ * The deadline class is a task class that has an additional parameter by.
+ */
+public class Deadline extends Task {
+
+    protected LocalDateTime by;
```

Comment:
Could there be a better name for this? Not sure if `by` might be too vague because it can have different meanings.